File: Easy/509-fibonacci-number/try-4.py
/*
 * Problem #509: Fibonacci Number
 * Difficulty: Easy
 * Submission: Try 4
 * status: Accepted
 * Language: python3
 * Date: 13/12/2024, 11:40:08
 * Link: https://leetcode.com/problems/fibonacci-number/
 */

class Solution:
    def fib(self, n: int) -> int:
        c=Counter({0:0,1:1})
        for i in range(2,n+1):
            c[i]=c[i-1]+c[i-2]
        return c[n]

    
        

File: Easy/268-missing-number/try-4.cpp
/*
 * Problem #268: Missing Number
 * Difficulty: Easy
 * Submission: Try 4
 * status: Accepted
 * Language: cpp
 * Date: 27/10/2025, 20:48:10
 * Link: https://leetcode.com/problems/missing-number/
 */

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = 0,n = nums.size();
        for(int i = 0;i <=n;i++){
            res ^= i;
        }
        for(int num : nums){
            res ^= num;
        }
        return res;
    }
};
